src/PrestaShopBundle/Form/Admin/Sell/Product/Pricing/RetailPriceType.php

```diff
@@ -130,6 +139,18 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             $ecotaxRate = new DecimalNumber('0');
         }
 
+        $selectedTaxRate = $this->taxComputer->getTaxRate(new TaxRulesGroupId($options['tax_rules_group_id']), new CountryId($this->contextCountryId));
+        $country = new Country($this->contextCountryId);
+        $taxGroupHelpText = 'Tax %1$s : %2$s%%';
+        $stateIsoCode = '';
+        if ($country->contains_states) {
+            $taxRules = \TaxRule::getTaxRulesByGroupId($this->contextLangId, $options['tax_rules_group_id']);
```

Comment:
We never use the localized values from this method so I don't think this function is adapted because:
- it returns much more data than we need
- it forces us to provide a languegId thus complexifying the form type for a useless info

Maybe there is another method which returns similar info (the purpose is to get the first state associated to a tax rule), and if there is none we can create a new method dedicated for this use, in which case it could even return the first one automatically and return type would be `int` which would simplify all the related codes even more

Or maybe it could be done in the more recent `TaxRulesGroupRepository` (see other comments about this service)